backend/src/email/middlewares/email-template.middleware.ts

```diff
@@ -67,6 +66,40 @@ const storeTemplate = async (req: Request, res: Response, next: NextFunction): P
     return next(err)
   }
 }
+
+/**
+ * Updates the campaign and email_messages table in a transaction, rolling back when either fails.
+ * For campaign table, the s3 meta data is updated with the uploaded file, and its validity is set to true.
+ * For email_messages table, existing records are deleted and new ones are bulk inserted.
+ * @param key 
+ * @param campaignId
+ * @param filename 
+ * @param records
+ */
+const updateCampaignAndMessages = async (
+  key: string,
+  campaignId: string,
+  filename: string,
+  records: MessageBulkInsertInterface[]): Promise<void> => {
+  let transaction
+
+  try {
+    transaction = await Campaign.sequelize?.transaction()
```

Comment:
Not sure if passing the transaction in functions is a good idea.

Reply:
I think it's fine having transaction as a param, just remember to take care of the case where it's undefined (ie. there isn't a transaction present)

Reply:
It seems like when `transaction` is `undefined` and used in a `sequelize` statement it is ignored, will try to google more.